src/gd.c

```diff
@@ -3539,20 +3539,12 @@ BGD_DECLARE(void) gdImageCopyResampled (gdImagePtr dst,
 				green /= alpha_sum;
 				blue /= alpha_sum;
 			}
-			/* Clamping to allow for rounding errors above */
-			if (red > 255.0) {
-				red = 255.0;
-			}
-			if (green > 255.0) {
-				green = 255.0;
-			}
-			if (blue > 255.0f) {
-				blue = 255.0;
-			}
-			if (alpha > gdAlphaMax) {
-				alpha = gdAlphaMax;
-			}
-			gdImageSetPixel(dst, x, y, gdTrueColorAlpha ((int) red, (int) green, (int) blue, (int) alpha));
+			/* Round up closest next channel value and clamp to max channel value */
+			red = red > 255.5 ? 255 : red+0.5;
```

Comment:
```suggestion
			red = red >= 255.5 ? 255 : red+0.5;
```

Reply:
both produces the same:

https://godbolt.org/z/ofr5znobc 

Basically the compiler realizes what we do, and ensure the val+0.5 clamping up:)

Reply:
Also good thing is we will be able to unroll this loop (no worry, not for 2.3 and for what I see gcc/clang and VC nicely parallelize it :)

Reply:
Try:
````.c
#include <stdio.h>

int main() {
    float red = 255.5;
    printf("%d\n", (int) (red > 255.5 ? 255 : red+0.5));
    printf("%d\n", (int) (red >= 255.5 ? 255 : red+0.5));
    return 0;
}
````
output:
````
256
255
````

Reply:
Oh! damned you right :)

Reply:
done :)

I will add a test for that specific case using only the gdTrueColorAlpha.